flake.nix

```diff
@@ -166,7 +166,15 @@
           snapFilter = mkFilter ".*snap$";
         in
         pkgs.lib.cleanSourceWith {
-          src = pkgs.lib.cleanSource ./.;
+          # Exclude paths that would trigger unnecessary rebuilds
+          src = pkgs.lib.cleanSourceWith {
+            src = pkgs.lib.cleanSource ./.;
+
+            filter = path: _type:
+              let basename = builtins.baseNameOf path; in
+
+              basename != ".github" && basename != ".vscode";
+          };
```

Comment:
Can this be merged into the `filter` function below somehow? It feels kind of weird to have two layers of `cleanSourceWith` for this.